### PR TITLE
Add documentation for decoupled support

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,8 @@ any C++ code.
   - [Usage](#usage)
     - [`initialize`](#initialize)
     - [`execute`](#execute)
+      - [Non-Decoupled mode](#non-decoupled-mode)
```
Comment (Tabrizian) [inline README.md:46]: How about renaming `Non-Decoupled mode` to `Default Mode`?

File: README.md [hunk 2]
```diff
@@ -275,9 +277,40 @@ below:
...
+    tensors or an error); an element cannot be None.
+
+
+Triton checks to ensure that these requirements on response list are
+satisfied and if not returns an error response for all inference requests.
+Upon return from the execute function all tensor data associated with the
+InferenceRequest objects passed to the function are deleted, and so
+InferenceRequest objects should not be retained by the Python model.
 
 In case one of the inputs has an error, you can use the `TritonError` object
```
Comment (Tabrizian) [inline README.md:315]: inputs -> requests

File: README.md [hunk 3]
```diff
@@ -302,6 +335,74 @@ class TritonPythonModel:
...
+
+* The return value for `execute` function in this mode should be None.
+
+Similar to above, in case one of the inputs has an error, you can use
+the `TritonError` object to set the error message for that specific
+request. After setting errors for an pb_utils.InferenceResponse
+object, use InferenceResponseSender.send() to send response with the
+error back to the user.
+
+##### Special Cases
```
Comment (Tabrizian) [inline README.md:375]: Special Cases -> Example Use Cases

File: README.md [hunk 4]
```diff
@@ -302,6 +335,74 @@ class TritonPythonModel:
         return responses
 ```
 
+
+#### Decoupled mode \[Beta\]
```
Comment (Tabrizian) [inline README.md:339]: Mention that in order to use this mode they should set the transaction policy to decoupled.

File: README.md [hunk 5]
```diff
@@ -302,6 +335,74 @@ class TritonPythonModel:
...
+full power of what can be acheived from decoupled API. Read
+[Decoupled Backends and Models](https://github.com/triton-inference-server/server/blob/main/docs/decoupled_models.md)
+for more details on how to host a decoupled model.
+
+##### Known Issues
+
+The support for decoupled models is still in beta and suffers
+from below known issues:
+
+* The decoupled mode does not support [GPU tensors](#interoperability-and-gpu-support).
```
Comment (Tabrizian) [inline README.md:402]: "The decoupled mode doesn't support `FORCE_CPU_ONLY_INPUT_TENSORS` parameter to be turned off. This means that the input tensors will always be in CPU."

On a separate bullet point:

"Currently, the InferenceResponseSender.send method only supports inference_response objects that contain only CPU tensors."

**Review by Tabrizian — CHANGES_REQUESTED** [review]
Could you please also add a link to the decoupled examples in the examples section?

File: README.md [hunk 6]
```diff
@@ -302,6 +335,74 @@ class TritonPythonModel:
...
+* Inferences on a decoupled model can not be run within [Business Logic Scripting](#business-logic-scripting).
```
[the remark this answers is not in the document]
Comment (tanmayv25) [reply README.md:403]: I think we should mention these limitation in both the places/

**Comment by mc-nv** [comment]
@tanmayv25 , @Tabrizian  could you please wrap it up

**Review by Tabrizian — APPROVED** [review]

**Pull request merged**